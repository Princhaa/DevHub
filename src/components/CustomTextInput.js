import React, { Component } from 'react'
import { View, Text, StyleSheet, TextInput } from 'react-native'

import constants from '../config/constants'

export default class CustomTextInput extends Component {

  constructor(props) {
    super(props)
    this.textInputRef = React.createRef()
    this.focus = this.focus.bind(this)
  }

  focus() {
    this.textInputRef.current.focus()
  }

  render() {
    const { children, ...otherProps } = this.props
    return (
      <View style={styles.container}>
        {children}
        <TextInput
          ref={this.textInputRef}
          style={styles.textInput}
          underlineColorAndroid={'transparent'}
          {...otherProps}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    width: constants.DEVICE_WIDTH * 0.9,
    borderColor: constants.GREY,
    borderRadius: 5,
    borderWidth: 0.3,
    height: 40,
    marginVertical: 5,
    flexDirection: 'row',
    alignItems: 'center'
  },

  textInput: {
    marginHorizontal: 10,
    flex: 2
  }
})
