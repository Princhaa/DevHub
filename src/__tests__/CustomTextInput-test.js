import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import CustomTextInput from '../components/CustomTextInput'

it('Renders CustomTextInput correctly', () => {
  const renderer = new ShallowRenderer()
  renderer.render(
    <CustomTextInput />
  )
  const tree = renderer.getRenderOutput()
  expect(tree).toMatchSnapshot()
})
