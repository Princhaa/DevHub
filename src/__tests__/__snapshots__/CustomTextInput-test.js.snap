// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CustomTextInput correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#B0B0B0",
      "borderRadius": 5,
      "borderWidth": 0.3,
      "flexDirection": "row",
      "height": 40,
      "marginVertical": 5,
      "width": 675,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    style={
      Object {
        "flex": 2,
        "marginHorizontal": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
</Component>
`;
